### DISCUSSION TITLE
Command to separate a clip into multiple clips based on linebreaks

### DISCUSSION THREAD

**Discussion by willswords** [hluk/CopyQ#2720] [Ideas]
Starting with text on the clipboard like this:
1 What is 4+4?
Eight
Two
Three
Twenty

I would like to run a command that puts each row of text into its own clipboard entry, e.g.
1 What is 4+4?
2 Eight
3 Two
4 Three
5 Twenty

This would make it much easier to paste each row separately into a tool for building quizzes. Is it possible to do this with CopyQ?

**Comment by willswords** [comment]
Or another approach would be to change the way paste works - only paste the top line until the linebreak of the clipboard and then delete that top line. Another tool does it this way: https://forum.keyboardmaestro.com/t/separate-each-line-on-system-clipboard-onto-its-own-clipboard/31082/15

**Comment by hluk** [comment]
To split the selected items, you can add the following command ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):

```ini
[Command]
Command="
    copyq:
    const text = str(input());
    const lines = text.split('\\n').reverse();
    for (const line of lines) {
        if (line)
            add(line);
    }
    select(0);"
Icon=
InMenu=true
Input=text/plain
Name=Split Lines
```

**Reply by willswords** [reply]
This is GREAT! Thank you so much for figuring this out for me! This is going to be very useful when moving multiple-choice questions into learning tools. 
One small correction - for some reason when you copy this command out of CopyQ, it changes \n to \\\\n. In line 5 of your text snippet above, it should be \n and not \\\\n. Once I figured this out, the script worked!

**Reply by hluk** [reply]
> One small correction - for some reason when you copy this command out of CopyQ, it changes \n to \\n. In line 5 of your text snippet above, it should be \n and not \\n.

You would normally paste the `[Command]` stuff to the command list in Command dialog (F6 shortcut) which would format the code properly (the code above is in `ini` configuration format, it requires `\` in values to be escaped/doubled).
